pkg/translator/prometheusremotewrite/metrics_to_prw_v2.go

```diff
@@ -131,9 +141,40 @@ func (c *prometheusConverterV2) addSample(sample *writev2.Sample, lbls []prompb.
 		off = c.symbolTable.Symbolize(l.Value)
 		buf = append(buf, off)
 	}
-	ts := writev2.TimeSeries{
+
+	sig := timeSeriesSignature(lbls)
+	ts := &writev2.TimeSeries{
 		LabelsRefs: buf,
 		Samples:    []writev2.Sample{*sample},
 	}
-	c.unique[timeSeriesSignature(lbls)] = &ts
+
+	// check if the time series is already in the unique map
+	if existingTS, ok := c.unique[sig]; ok {
```

Comment:
[Nit] I think it's better to follow the more robust approach of the PRW v1 code, and check whether `existingTS` is nil because you then kill two birds with one stone (you check both whether `existingTS` is nil and whether it exists in the map):
```suggestion
	if existingTS := c.unique[sig]; existingTS != nil {
```